igvc_rviz_plugins/include/igvc_rviz_plugins/sensor_panel.h

```diff
@@ -0,0 +1,71 @@
+#ifndef SENSORPANEL_H
+#define SENSORPANEL_H
+
+#include <ros/ros.h>
+#include <rviz/panel.h>
+#include <sensor_msgs/Image.h>
+#include <sensor_msgs/Imu.h>
+#include <sensor_msgs/NavSatFix.h>
+#include <sensor_msgs/PointCloud2.h>
+#include <QLabel>
+#include <QObject>
+#include <QTimer>
+#include <string>
+
+#define INTERVAL 250//in milliseconds
+
+/*
+ * All of our panels need to be under the igvc_rviz_plugins namespace.
+ */
+namespace igvc_rviz_plugins
+{
+/*
+ * Each panel is a subclass of rviz::Panel
+ */
+class sensor_panel : public rviz::Panel
+{
+  /*
+   * rviz is based on QT, so our panels need to be QObjects, which requires this macro keyword.
+   */
+  Q_OBJECT
+public:
+  /**
+   * This is a standard QWidget constructor.
+   * @param parent The parent widget, which will be responsible for the lifetime of this widget.
+   */
+  sensor_panel(QWidget *parent = 0);
+
+public slots:
+  void timer();
+
+protected:
+  /*
+   * Be sure to make any publishers / subscribers members of the class. This will keep them alive throughout
+   * the lifetime of the widget.
+   */
+  ros::Subscriber imu_sub;
+  ros::Subscriber lidar_sub;
+  ros::Subscriber gps_sub;
+  ros::Subscriber cam_center_sub;
+
+  QTimer *sensor_timer;
+  std::map<std::string, QLabel *> labels;// map of pointers to labels
+  bool *isActive; // list of activity for sensors
+
+  /**
```

Comment:
I think we can remove these comments or make them useful, personal discretion here.

Reply:
Applies to all comments

Reply:
Im fixing the comments still